aten/src/ATen/native/cuda/ForeachPointwiseOp.cu

```diff
@@ -46,37 +89,53 @@ void foreach_pointwise_op_(TensorList input, TensorList tensors1, TensorList ten
-        !can_use_fast_route(tensors1, tensors2) ||                                                                             \
-        !can_use_fast_route(input, tensors1)) {                                                                                \
-        return at::native::foreach_tensor_##NAME##_slow(input, tensors1, tensors2, scalar);                                    \
-    }                                                                                                                          \
-                                                                                                                               \
-    return foreach_pointwise_op<OP>(input, tensors1, tensors2, scalar);                                                        \
-}                                                                                                                              \
-                                                                                                                               \
-void foreach_tensor_##NAME##_cuda_(TensorList input, TensorList tensors1, TensorList tensors2, Scalar scalar) {                \
-    TORCH_CHECK(input.size() > 0, "Tensor list must have at least one tensor.");                                               \
-    TORCH_CHECK(input.size() ==  tensors1.size(), "Tensor lists must be of the same length.");                                 \
-    TORCH_CHECK(tensors1.size() ==  tensors2.size(), "Tensor lists must be of the same length.");                              \
-                                                                                                                               \
-    if (!can_use_fast_route(input, scalar) ||                                                                                  \
-        !can_use_fast_route(tensors1, tensors2) ||                                                                             \
-        !can_use_fast_route(input, tensors1)) {                                                                                \
-        return at::native::foreach_tensor_##NAME##_slow_(input, tensors1, tensors2, scalar);                                   \
-    }                                                                                                                          \
-                                                                                                                               \
-    foreach_pointwise_op_<OP>(input, tensors1, tensors2, scalar);                                                              \
+#define FOREACH_POINTWISE_OP_SCALAR(NAME, OP)                                                                                         \
+std::vector<Tensor> foreach_tensor_##NAME##_scalar_cuda(TensorList input, TensorList tensors1, TensorList tensors2, Scalar scalar) {  \
```

Comment:
nit: (no need to change) PyTorch doesn't really have code style guides. However, for future reference, keeping the line width to < 100 helps, especially during code review. Right now I have to maximize my window to full screen to see the correct indentation:

![image](https://user-images.githubusercontent.com/5652049/95919674-04a59b80-0d7c-11eb-9fd9-f4cfa35c7a6d.png)